### ISSUE TITLE
Cypress doesn't work with Webpack 5 

### ISSUE THREAD

**Issue by ArielGueta** [nrwl/nx#5713]
After upgrading to Webpack 5, we get this error:

> Automatic publicPath is not supported in this browser

It seems that you need to update `@cypress/webpack-preprocessor` to latest version. 

https://github.com/cypress-io/cypress/issues/8900

**Comment by vincentpalita** [comment]
I confirm that, adding
`"@cypress/webpack-preprocessor": "^5.9.0"`
to the resolutions section of our package.json file fixed the issue for us.

**Comment by vincentpalita** [comment]
I have in fact sometime the error you mention:
> Automatic publicPath is not supported in this browser
But some specs are passing and some not, will try to find out why.

**Comment by markhus-aurelius** [comment]
> I confirm that, adding
> `"@cypress/webpack-preprocessor": "^5.9.0"`
> to the resolutions section of our package.json file fixed the issue for us.

Cypress hangs on 'tests loading' for me when I force this resolution. But at least the publicPath error is not showing, hah.

> I have in fact sometime the error you mention:
> 
> > Automatic publicPath is not supported in this browser
> 
> But some specs are passing and some not, will try to find out why.

Otherwise also getting the issue where some tests are fine other show the `Automatic publicPath is not supported in this browser` error.

Did you find a solution for this?

**Comment by vincentpalita** [comment]
@markhus-aurelius as of now I haven't had a chance to find why. I am guessing a library that some of our specs are depending on has a problem when webpack is requiring it.
I'll update you as soon as I find out.

**Comment by ubergeoff** [comment]
I had this issue where some cypress tested passed while other failed with the error:

```
> Automatic publicPath is not supported in this browser
```

I found that I was making use of tsconfig paths within my Cypress test classes:

```typescript
import { SbgConstants } from '@sbg-web/sbg-common';
```

If I changed these references to absolute paths. Then everything started working again:

```
// tslint:disable-next-line: nx-enforce-module-boundaries
import { SbgConstants } from '../../../../libs/sbg-common/src/lib/constants/sbg-constants';
```

**Comment by vincentpalita** [comment]
@markhus-aurelius, @ubergeoff we seem to have found a solution by removing the use of preprocessTypescript function provided by Nx and just install the 2 following dev dependencies: 
"@cypress/webpack-dev-server"
"webpack-dev-server"

since cypress v4.4.0 you don't need any filePreprocessor to use Typescript to write your specs, nonetheless you will need the dependencies listed above.

@FrozenPandaz I am pinging you here because I saw this PR that was merged in April, it might be the cause of this issue: https://github.com/cypress-io/cypress/pull/15839

My guess is that maybe the filePreprocessor provided by Nx could be deprecated, there was already a thread about that: https://github.com/nrwl/nx/issues/3037

Or we need to find a way to make the publicPath work when using webpack 5.

**Comment by ubergeoff** [comment]
Thank you @vincentpalita - yes I can confirm that after installing:

```
"@cypress/webpack-dev-server": "^1.3.1",
```

And updating my `plugins/index.js` file to the below also fixes this issue...

```typescript
module.exports = function (on, config) {
    //on('file:preprocessor', config);
    // `on` is used to hook into various events Cypress emits
    // `config` is the resolved Cypress config
};
```

**Comment by markhus-aurelius** [comment]
@vincentpalita Absolute legend mate, this got it working for me. Appreciate the help.

**Comment by docwhite** [comment]
@markhus-aurelius @vincentpalita Do you speak about React applications or Angular applications? If I do what you suggest instead of the publicPath error message I get this one (I suspect it's perhaps because I use Angular?):

```
Error: Webpack Compilation Error
/Users/ramon/Watchity/frontend/libs/shared/ui/src/index.ts 15:9
Module parse failed: Unexpected token (15:9)
File was processed with these loaders:
 * ../../../../Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/ts-loader/index.js
You may need an additional loader to handle the result of these loaders.
| export * from './lib/store/ui.actions';
| export * from './lib/store/ui.selectors';
> export * as fromUi from './lib/store/ui.reducer';
| // Service.
| export * from './lib/services/toast.service';
 @ /Users/ramon/Watchity/frontend/libs/shared/login/src/lib/interceptors/login.interceptor.ts 13:0-62 86:40-63 90:40-63 94:40-63
 @ /Users/ramon/Watchity/frontend/libs/shared/login/src/index.ts
 @ ./src/support/modules/login/login.commands.ts
 @ ./src/support/index.ts
    at Watching.handle [as handler] (/Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/webpack-preprocessor/dist/index.js:172:23)
    at /Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/Watching.js:99:9
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/tapable/lib/Hook.js:154:20)
    at Watching._done (/Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/Watching.js:98:28)
    at /Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/Watching.js:73:19
    at Compiler.emitRecords (/Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/Compiler.js:499:39)
    at /Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/Watching.js:54:20
    at /Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/Compiler.js:485:14
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/tapable/lib/Hook.js:154:20)
    at /Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/Compiler.js:482:27
    at /Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/neo-async/async.js:2818:7
    at done (/Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/neo-async/async.js:3522:9)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/tapable/lib/Hook.js:154:20)
    at /Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/Compiler.js:464:33
    at /Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/graceful-fs/graceful-fs.js:111:16
    at /Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/enhanced-resolve/node_modules/graceful-fs/graceful-fs.js:61:14
    at /Users/ramon/Library/Caches/Cypress/7.5.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/graceful-fs/graceful-fs.js:45:10
    at FSReqCallback.oncomplete (fs.js:171:23)
```

**Comment by paustint** [comment]
I am posting here just in case anyone stumbles on this thread because of the error message, even though the cause of the error was different in my case.

After upgrading to webpack 5, my react app would not load with the error `Automatic publicPath is not supported in this browser`.

The issue for me was that I was referencing `url` in css - example `background-image: url('assets/images/split-grip-vertical.png');` and apparently that causes issues with `mini-css-extract-plugin` (or something else) and I found this thread: https://stackoverflow.com/questions/64294706/webpack5-automatic-publicpath-is-not-supported-in-this-browser.

I fixed it by setting `outputPath` to an empty string in my webpack config.

Here is a simplified example:

```javascript
const getWebpackConfig = require('@nrwl/react/plugins/webpack');

module.exports = (config) => {
  config = getWebpackConfig(config);
  config.output.publicPath = '';
  return config;
};
```
